stl/inc/xutility

```diff
@@ -376,17 +375,14 @@ struct _Ref_fn { // pass function object by value as a reference
 };
 
 template <class _Fn>
-_INLINE_VAR constexpr bool _Pass_functor_by_value_v = conjunction_v<bool_constant<sizeof(_Fn) <= sizeof(void*)>,
-    is_trivially_copy_constructible<_Fn>, is_trivially_destructible<_Fn>>;
-
-template <class _Fn, enable_if_t<_Pass_functor_by_value_v<_Fn>, int> = 0> // TRANSITION, if constexpr
-constexpr _Fn _Pass_fn(_Fn _Val) { // pass functor by value
-    return _Val;
-}
-
-template <class _Fn, enable_if_t<!_Pass_functor_by_value_v<_Fn>, int> = 0>
-constexpr _Ref_fn<_Fn> _Pass_fn(_Fn& _Val) { // pass functor by "reference"
-    return {_Val};
+constexpr auto _Pass_fn(_Fn& _Func) noexcept {
+    constexpr bool _Pass_by_value = conjunction_v<bool_constant<sizeof(_Fn) <= sizeof(void*)>,
+        is_trivially_copy_constructible<_Fn>, is_trivially_destructible<_Fn>>;

```

Comment:
I think this is fine as-is. While we support Clang as a first-class citizen (for feature support, testing, etc.), it is certainly the case that MSVC's STL is used with the MSVC compiler much more often. In the case of the type traits optimizations, when faced with a choice between micro-optimizing throughput for MSVC or Clang, I believe that we should continue to assume the presence of MSVC's type traits optimizations. Clang can always implement similar optimizations if they would like to improve throughput as we did (this would also benefit libc++, libstdc++, and user code - everyone loves type traits).

It's also worth noting that this is preserving the existing code where `_Pass_functor_by_value_v` used `conjunction_v` and that was never an observed issue for Clang.

Reply:
I expect this change to increase MSVC's compile times of large programs by a few microseconds (we optimize `is_trivially_destructible_v` and `is_trivially_copy_constructible_v`), and reduce clang compile times of the same by a few milliseconds. I'd honestly be surprised if we could devise a benchmark that demonstrated an observable difference in compile times on either compiler.

That said, I'd prefer the suggested change for simplicity - but not without universal approval. There's nothing here that's worth our time to argue the decision.

Reply:
yeah no worries, just drop it